src/main/java/com/johnstarich/moviematcher/controllers/MovieService.java

```diff
@@ -57,7 +57,9 @@ public void initService() {
 		Route movieRoute = (request, response) -> {
 			String movieId = request.params("id");
 			System.out.println("Looked up movie with ID: " + movieId);
-			return new Movie(new ObjectId(movieId)).load(Movie.class);
+			Optional<Movie> result = new Movie(new ObjectId(movieId)).load(Movie.class);
+			if(! result.isPresent()) throw new HttpException(HttpStatus.NOT_FOUND, "Movie not found with ID: "+movieId);
```

Comment:
I addressed the other notes left below. 
Regarding using a ClientFacingHttpException, the pull request containing ClientFacingHttpException has not yet been merged.